cmd/eksctl-anywhere/cmd/validatecreatecluster.go

```diff
@@ -0,0 +1,140 @@
+	)
+	validateCmd.Flags().StringVar(&valOpt.tinkerbellBootstrapIP, "tinkerbell-bootstrap-ip", "", "Override the local tinkerbell IP in the bootstrap cluster")
+
+	if err := validateCreateCmd.MarkFlagRequired("filename"); err != nil {
+		log.Fatalf("Error marking flag as required: %v", err)
+	}
+}
+
+func (valOpt *validateOptions) validateCreateCluster(cmd *cobra.Command, _ []string) error {
+	ctx := cmd.Context()
+
+	clusterSpec, err := cluster.NewSpecFromClusterConfig(valOpt.fileName, version.Get())
+	if err != nil {
+		return err
+	}
+
+	if clusterSpec.Config.Cluster.Spec.DatacenterRef.Kind == v1alpha1.TinkerbellDatacenterKind {
+		flag := cmd.Flags().Lookup(TinkerbellHardwareCSVFlagName)
+
+		// If no flag was returned there is a developer error as the flag has been removed
+		// from the program rendering it invalid.
+		if flag == nil {
+			panic("'hardwarefile' flag not configured")
+		}
+
+		if !viper.IsSet(TinkerbellHardwareCSVFlagName) || viper.GetString(TinkerbellHardwareCSVFlagName) == "" {
+			return fmt.Errorf("required flag \"%v\" not set", TinkerbellHardwareCSVFlagName)
+		}
+
+		if !validations.FileExists(valOpt.hardwareCSVPath) {
+			return fmt.Errorf("hardware config file %s does not exist", valOpt.hardwareCSVPath)
+		}
+	}
+
+	cliConfig := buildCliConfig(clusterSpec)
+	dirs, err := valOpt.directoriesToMount(clusterSpec, cliConfig)
+	if err != nil {
+		return err
+	}
+
+	tmpPath, err := os.MkdirTemp("./", "tmpValidate")
+	if err != nil {
+		return err
+	}
+	deps, err := dependencies.ForSpec(ctx, clusterSpec).WithExecutableMountDirs(dirs...).
+		WithWriterFolder(tmpPath).
+		WithKubectl().
+		WithProvider(valOpt.fileName, clusterSpec.Cluster, false, valOpt.hardwareCSVPath, true, valOpt.tinkerbellBootstrapIP).
+		WithGitOpsFlux(clusterSpec.Cluster, clusterSpec.FluxConfig, cliConfig).
+		Build(ctx)
+	if err != nil {
+		deps.Writer.CleanUp()
+		return err
+	}
+	defer close(ctx, deps)
+
+	var cluster *types.Cluster
+	if clusterSpec.ManagementCluster == nil {
+		cluster = &types.Cluster{
+			Name:           clusterSpec.Cluster.Name,
+			KubeconfigFile: kubeconfig.FromClusterName(clusterSpec.Cluster.Name),
+		}
+	} else {
+		cluster = &types.Cluster{
+			Name:           clusterSpec.ManagementCluster.Name,
+			KubeconfigFile: clusterSpec.ManagementCluster.KubeconfigFile,
+		}
+	}
+
+	validationOpts := &validations.Opts{
+		Kubectl: deps.Kubectl,
+		Spec:    clusterSpec,
+		WorkloadCluster: &types.Cluster{
+			Name:           clusterSpec.Cluster.Name,
+			KubeconfigFile: kubeconfig.FromClusterName(clusterSpec.Cluster.Name),
+		},
+		ManagementCluster: cluster,
+		Provider:          deps.Provider,
+		CliConfig:         cliConfig,
+	}
+
+	createValidations := createvalidations.New(validationOpts)
+
+	docker := executables.BuildDockerExecutable()
+
+	commandVal := commandvalidations.New(clusterSpec, deps, createValidations, docker)
+	err = commandVal.ValidateCreate(ctx)
+
+	cleanup(deps, &err)
+	deps.Writer.CleanUp()
```

Comment:
doesn't `cleanup` do this?

Reply:
Ended up removing both and just using cleanupDirectory()